control-plane-operator/controllers/hostedcontrolplane/oapi/deployment_test.go

```diff
@@ -0,0 +1,235 @@
+package oapi
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+	configv1 "github.com/openshift/api/config/v1"
+	hyperv1 "github.com/openshift/hypershift/api/hypershift/v1beta1"
+	"github.com/openshift/hypershift/control-plane-operator/controllers/hostedcontrolplane/manifests"
+	"github.com/openshift/hypershift/support/config"
+	"github.com/openshift/hypershift/support/util"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Ensure certain deployment fields do not get set
+func TestReconcileOpenshiftAPIServerDeploymentNoChanges(t *testing.T) {
+
+	imageName := "oapiImage"
+	// Setup expected values that are universal
+
+	// Setup hypershift hosted control plane.
+	targetNamespace := "test"
+	oapiDeployment := manifests.OpenShiftAPIServerDeployment(targetNamespace)
+	hcp := &hyperv1.HostedControlPlane{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "hcp",
+			Namespace: targetNamespace,
+		},
+	}
+	hcp.Name = "name"
+	hcp.Namespace = "namespace"
+	ownerRef := config.OwnerRefFrom(hcp)
+	serviceServingCA := manifests.ServiceServingCA(hcp.Namespace)
+
+	testCases := []struct {
+		cm                    corev1.ConfigMap
+		auditConfig           *corev1.ConfigMap
+		deploymentConfig      config.DeploymentConfig
+		additionalTrustBundle *corev1.LocalObjectReference
+		clusterConf           *hyperv1.ClusterConfiguration
+	}{
+		// empty deployment config
+		{
+			cm: corev1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-oapi-config",
+					Namespace: targetNamespace,
+				},
+				Data: map[string]string{"config.yaml": "test-data"},
+			},
+			auditConfig:           manifests.OpenShiftAPIServerAuditConfig(targetNamespace),
+			deploymentConfig:      config.DeploymentConfig{},
+			additionalTrustBundle: &corev1.LocalObjectReference{Name: "test-trust-bundle"},
+			clusterConf:           nil,
+		},
+	}
+	for _, tc := range testCases {
+		g := NewGomegaWithT(t)
+		expectedTermGraceSeconds := oapiDeployment.Spec.Template.Spec.TerminationGracePeriodSeconds
+		oapiDeployment.Spec.MinReadySeconds = 60
+		expectedMinReadySeconds := oapiDeployment.Spec.MinReadySeconds
```

Comment:
it seems you are carrying oapiDeployment over all test cases loop which would let each test have side effects on each other.

Reply:
humm I think I will create a manual backport, from a fresh and rebased release-4.15 branch and do the modifications there.